### ISSUE TITLE
Issue when using vroom_fwf() to read in large .dat.gz files

### ISSUE THREAD

**Issue by stephhazlitt** [r-lib/vroom#112]
I am using vroom_fwf() to read in .data.gz files. This works on some files, however on some files, which appear to be the larger files (< ~85M), some of the last rows are garbled. Trying to view the garbled rows in the RStudio IDE or using the unstable object in subsequent code (like trying to filter with `dplyr`) crashes RStudio. The same compressed files are read in fine with `readr::read_fwf` and the uncompressed .dat counterparts are read in fine with `vroom_fwf()`.

Creating a reprex is a challenge due to both the data and computing environment where this behaviour is observed, however please let me know if there is anything I can do to provide more information.

The machine is running R v3.6.0, RStudio v1.2.1335, and vroom v0.1.0.

**Comment by jimhester** [comment]
How are you specifying the column positions for the fixed with file? Are you using `fwf_empty()` or a different method? Is the data in any way different at the end of the file?

**Comment by jimhester** [comment]
Another thing to try is to reduce the `VROOM_CONNECTION_SIZE` [environment variable](https://github.com/r-lib/vroom#environment-variables) to see if the problem then happens with smaller files. The default is `131072`, so you could try something like `16384`

**Comment by stephhazlitt** [comment]
OK, I'll try that ☝️ and report back. The general approach looks like this:

```
vroom_fwf(file,
col_positions = 
vroom::fwf_positions(
c(1, 4, 6),
c(3, 5, 6),
c("var1", "var2", "var3")),
col_types = c("ccc"))
```

**Comment by stephhazlitt** [comment]
>  Is the data in any way different at the end of the file?

I don't _think_ so, however it is hard to look. Reading in a problem file with both `readr::read_fwf` and `vroom::vroom_fwf` generates objects with the same number of rows. When I try to find the row where things stop working in the `vroom` object, through the viewer pane or programmatically, I crash the IDE :bomb:.

**Comment by jimhester** [comment]
You could try using `vroom::vroom_lines()` to look at the raw data of the lines, it should work without crashing.

**Comment by stephhazlitt** [comment]
> You could try using `vroom::vroom_lines()` to look at the raw data of the lines, it should work without crashing.

I only have v1.0.0 at this time, but I'll see if I can get the latest release with `vroom_lines()`.

**Comment by stephhazlitt** [comment]
> Another thing to try is to reduce the `VROOM_CONNECTION_SIZE` [environment variable](https://github.com/r-lib/vroom#environment-variables) to see if the problem then happens with smaller files. The default is `131072`, so you could try something like `16384` or smaller

Reporting back. I (_think_) I dropped the connection size and re-ran on a smallish file that was working, and it still worked. 

```
Sys.getenviron("VROOM_CONNECTION_SIZE")
```
got me `[1] ""`, so I ran this

```
Sys.setenviron("VROOM_CONNECTION_SIZE" = 16384)
```
which gave me this `"131072"`. Then I re-ran the code chunk, no issues. I tried some smaller values as well. Hoping this was what you meant 😳 .

**Comment by stephhazlitt** [comment]
I now have `vroom v1.0.1` and ran `vroom_lines()` with one of the troublesome `.dat.gz` files (which work fine w/ `vroom_fwf()`  if I unpack them to .dat manually first). I get the error `Error: R character strings are limited to 2^31-1 bytes`. I figured out where the "bad egg" lines start and when I run `vroom_lines(data, skip = xxxx)` to jump straight to these rows  I get `Error: The size of the connection buffer (131072) was not large enough to fit a complete line: Increase it by setting Sys.setenv("VROOM_CONNECTION_SIZE")`. I have tried increasing the connection buffer with no success -- how big can that reasonably go?

**Comment by jimhester** [comment]
You could try using just `readr::read_lines()` then to look at the lines, the file should be small enough to fit in memory.

**Comment by stephhazlitt** [comment]
Thanks @jimhester, that suggestion worked to look at the data but sadly I cannot see anything that looks 🐟 y? As a work-a-around for these files I am using `R.utils::gunzip` before vrooming. I have requested some synthetic data to see if I can reproduce this in an open environment.

**Comment by jimhester** [comment]
If you create a new file with just the lines from the middle that are causing the issue does it still happen? Or does it only happen when they are at the place in the file? e.g. something like

```r
lines <- readr::read_lines("file.gz")
# Let's say the issue starts on line 10000, we will take the first line (assuming it is a header) and then the lines after 10,000+
new_lines <- lines[c(1, 10000:length(lines))]
readr::write_lines("new_file.gz")
vroom::vroom_fwf("new_file.gz", my_col_positions)
```

What about if you just re-arrange the lines?
```r
new_lines <- sample(lines)
```

Or move the problematic lines to the front of the file?
```r
new_lines <- lines[c(10001:11000, 1:10000, 11001:length(lines))]
```

**Comment by jimhester** [comment]
I think there is a good chance that a66437a fixes this issue, if it does not we can reopen it.

**Comment by stephhazlitt** [comment]
Tx @jimhester. I will test once that fix hits CRAN and report back (I don't have access to the dev version in the 🌎 where the files live).

**Comment by stephhazlitt** [comment]
Hi @jimhester &mdash; I can confirm that this issue is resolved with `vroom` 1.0.2. Tx so much  🙌 .

**Comment by petyaracz** [comment]
I have this error w/ a 2.5gb gz file using `vroom_lines`, when I read input and skip lines. Given the size of the dataset I'm unsure how to create a reprex (but I'm happy to work on it!).

vroom_1.3.2
R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6
